### PR TITLE
Remove hardcoded styling in plot_histogram

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
`plot_histogram` was hardcoding things like bar colors, font and figure sizes, etc, that were preventing Matplotlib style files from being faithfully applied to the images.

Here we remove the hardcoded values, allowing complete customization following the loaded `stylelib`.  Users wishing to modify the default settings further can either make their own styles, or take advantage of the fact that one can pass an `Axes` object, giving complete control of the customization post-generation.

### Details and comments
The bar colors are now sourced from the `rcParams` which are in turn set by the style.  Most hard-coded options have been removed save for the rotation of the x-axis labels that need to be rotated.  I have also disabled `bar_labels` by default as for anything more than 2-3Q, the labels are unreadable, and unnecessary given the y-axis is also labeled and the data is included in the `Counts` themselves.

This allows one to make figures that look like:

image

or 

image

easily

### CODE REVIEW

**Comment by nonhermitian** [comment]
This should wait for #8762 to be merged so that I can refactor on top of those changes.

File: releasenotes/notes/histogram-style-03807965c3cc2e8a.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+upgrade:
+  - |
+    Removes the hard-coding of style options for ``plot_histogram``.  This allows
```
Comment (1ucian0) [inline releasenotes/notes/histogram-style-03807965c3cc2e8a.yaml]:
```suggestion
    Removes the hard-coding of style options for :func:`~.plot_histogram`.  This allows
```

File: qiskit/visualization/counts_visualization.py [hunk 2]
```diff
@@ -70,13 +70,13 @@ def _is_deprecated_data_format(data) -> bool:
...
     data,
-    figsize=(7, 5),
+    figsize=None,
     color=None,
     number_to_keep=None,
     sort="asc",
     target_string=None,
     legend=None,
-    bar_labels=True,
+    bar_labels=False,
```
Comment (1ucian0) [inline qiskit/visualization/counts_visualization.py]: was this change intentional?

**Review by 1ucian0 — COMMENTED** [review]
Thanks a lot for this PR. The feature makes sense to me. Just a small comment on `bar_labels` that I dont get.

File: qiskit/visualization/counts_visualization.py (hunk 2 above)
Comment (nonhermitian) [reply qiskit/visualization/counts_visualization.py]: Ahh, this was because the labels are redundant given the y-axis labels, and easily becomes messy when the number of bit-strings is large, and the figure dims are not set properly (if possible).

Comment (1ucian0) [reply qiskit/visualization/counts_visualization.py]: is the change in the default worthy an `upgrade` note in the release note?

Comment (1ucian0) [reply qiskit/visualization/counts_visualization.py]: Undoing this change, as it is the only blocker. For sure, it can be done in 2.0 as a breaking change. 

```suggestion
    bar_labels=True,
```

**Review by 1ucian0 — APPROVED** [review]
